### PR TITLE
Adds migration files and models with tests

### PR DESCRIPTION
**Pull request by brandomoreno06**
**Inclusions:** 

Migration files and models with spec test for tables:

1. courses
2. subjects
3. rc_teachers
3. rc_courses
4. lessons

**Screenshot**
image

### CODE REVIEW

File: spec/models/lesson_spec.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
+
+RSpec.describe Lesson, type: :model do
+  context 'when validating presence' do
+    it { is_expected.to validate_presence_of(:name) }
+    it { is_expected.to validate_presence_of(:teacher_subject_id) }
+    it { is_expected.to validate_presence_of(:rc_course_id) }
+    it { is_expected.to validate_presence_of(:price) }
```
Comment (analyn-cajocson) [inline spec/models/lesson_spec.rb:8]: No need to validate the foreign key. You already have the test for the associations

File: spec/models/rc_course_spec.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+require 'rails_helper'
+
+RSpec.describe RcCourse, type: :model do
+  context 'when validating presence' do
+    it { is_expected.to validate_presence_of(:review_center_id) }
+    it { is_expected.to validate_presence_of(:course_id) }
```
Comment (analyn-cajocson) [inline spec/models/rc_course_spec.rb:6]: No need to validate the foreign key. You already have the test for the associations

File: spec/models/rc_teacher_spec.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+require 'rails_helper'
+
+RSpec.describe RcTeacher, type: :model do
+  context 'when validating presence' do
+    # it { is_expected.to validate_presence_of(:user_id) }
+    it { is_expected.to validate_presence_of(:review_center_id) }
```
Comment (analyn-cajocson) [inline spec/models/rc_teacher_spec.rb:6]: No need to validate the foreign key. You already have the test for the associations

File: spec/models/subject_spec.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+require 'rails_helper'
+
+RSpec.describe Subject, type: :model do
+  context 'when validating presence' do
+    it { is_expected.to validate_presence_of(:name) }
+    it { is_expected.to validate_presence_of(:course_id) }
```
Comment (analyn-cajocson) [inline spec/models/subject_spec.rb:6]: No need to validate the foreign key. You already have the test for the associations

File: spec/models/teacher_subject_spec.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+require 'rails_helper'
+
+RSpec.describe TeacherSubject, type: :model do
+  context 'when validating presence' do
+    it { is_expected.to validate_presence_of(:rc_teacher_id) }
+    it { is_expected.to validate_presence_of(:subject_id) }
```
Comment (analyn-cajocson) [inline spec/models/teacher_subject_spec.rb:6]: No need to validate the foreign key. You already have the test for the associations

**Review by Stephdajon — APPROVED** [review]

**Pull request merged**